### PR TITLE
FB-1895: Implement DateTimeFromParts

### PR DESCRIPTION
**Pull request by rachithrr**
this uses the tiem.date() function

### CODE REVIEW

**Comment by rachithrr** [comment]
Well, I have to add limits to the time ranges. For example, dax expects the year range [0 , 9999]. I'll update the code and re request to review.

**Review by lorycloutier — DISMISSED** [review]
This looks correct as far as I can tell.

File: sql3/test/defs/defs_date_functions.go [hunk 1]
```diff
@@ -19,48 +22,56 @@ var datePartTests = TableTest{
 	),
 	SQLTests: []SQLTest{
 		{
+			name: "DatePartIncorrectParamsCount",
```
Comment (vkrishnanfb) [inline sql3/test/defs/defs_date_functions.go:25]: Didn't know test case names can be defined. I merged a whole bunch of tests without names, will remember to fix them when I touch those files next time.

File: sql3/planner/inbuiltfunctionsdate.go [hunk 2]
```diff
@@ -150,6 +166,34 @@ func (n *callPlanExpression) EvaluateDatepart(currentRow []interface{}) (interfa
 
 }
 
+// EvaluateDateTimeFromParts evaluates the call to date_time_from_parts. This uses the base time.Date() function.
+func (n *callPlanExpression) EvaluateDateTimeFromParts(currentRow []interface{}) (interface{}, error) {
+	timestamps := make([]int, len(n.args))
```
Comment (vkrishnanfb) [inline sql3/planner/inbuiltfunctionsdate.go:206]: I see we always want to create date and time. A date only variation accepting just YYYY, MM, DD would be nice to have.

File: sql3/planner/inbuiltfunctionsdate.go [hunk 3]
```diff
@@ -150,6 +166,34 @@ func (n *callPlanExpression) EvaluateDatepart(currentRow []interface{}) (interfa
...
+		}
+		val, ok := coercedValue.(int64)
+		if !ok {
+			return nil, sql3.NewErrInternalf("unable to convert value")
+		}
+		timestamps[i] = int(val)
+	}
+
+	dt := time.Date(timestamps[0], time.Month(timestamps[1]), timestamps[2], timestamps[3], timestamps[4], timestamps[5], timestamps[6]*1000*1000, time.UTC)
+	if dt.Year() < 0 || dt.Year() > 9999 {
```
Comment (vkrishnanfb) [inline sql3/planner/inbuiltfunctionsdate.go:226]: Silent normalization of out of range values for Month and below should be highlighted in the documentation. May be an example showing October 32 would convert to November 1 would nicely illustrate this.

**Comment by vkrishnanfb** [comment]
Looks good to me.

**Review by vkrishnanfb — APPROVED** [review]

File: sql3/planner/inbuiltfunctionsdate.go (hunk 2 above)
Comment (rachithrr) [reply sql3/planner/inbuiltfunctionsdate.go:231]: thats true. We can add more going forward.

File: sql3/planner/inbuiltfunctionsdate.go (hunk 3 above)
Comment (rachithrr) [reply sql3/planner/inbuiltfunctionsdate.go:251]: Yes. I'll make sure to note this.

**Pull request merged**